draft-ietf-ppm-dap.md

```diff
@@ -1473,43 +1500,74 @@ prepared:
+  collection as described in {{collect-flow}} and replies to the Leader with a
+  `PrepareStep` in the `finished` state.
+* An updated VDAF state and preparation message, denoted
+  `(prep_state, prep_msg)`, in which case the Helper replies to the Leader with
+  a `PrepareStep` in the `continued` state containing `prep_msg`.
+
+If the `round` in the Leader's request is equal to the Helper's current round
+(i.e., this is not the first time the Leader has sent this request), then the
+Helper responds with the current round's prepare messages. The Helper MUST
+verify that the contents of the `AggregationJob` are identical to the previous
+message (see {{aggregation-round-skew-recovery}}).
+
+If the Leader's `round` is behind or more than one round ahead of the Helper's
+current round, then the Helper MUST abort with an error of type `roundMismatch`.
+
+If successful, the Helper responds to the Leader with HTTP status 200 OK, media
+type `application/dap-aggregation-job` and a body consisting of an
+`AggregationJob` (see {{aggregation-helper-init}}) compiled from the stepped
+`PrepareStep`s.
+
+#### Recovering From Round Skew {#aggregation-round-skew-recovery}
+
+`AggregationJob` messages contain a `round` field, allowing Aggregators to
+ensure that their peer is on an expected round of the VDAF preparation
+algorithm. In particular, the intent is to allow recovery from a scenario where
+the Helper successfully advances from round `n` to `n+1`, but its
+`AggregationJob` response to the Leader gets dropped due to something like a
+transient network failure. The Leader could then resend the request to have the
+Helper advance to round `n+1` and the Helper should be able to retransmit the
+`AggregationJob` that was previously dropped. To make that kind of recovery
+possible, Aggregator implementations SHOULD checkpoint preparation state to
+durable storage such that Leaders can re-construct continuation requests and
+Helpers can re-construct continuation responses as needed.
+
+When implementing a round skew recovery strategy, Helpers SHOULD ensure that the
+Leader's `AggregationJob` message did not change when it was re-sent (i.e., the
+two messages must contain the same set of report IDs and prepare messages). This
+prevents the Leader from re-winding an aggregation job and re-running a round
+with different parameters.
+
+[[OPEN ISSUE: Allowing the Leader to "rewind" aggregation job state of an may
+allow an attack on privacy. For instance, if the VDAF verification key changes,
+the preparation shares in the Helper's response would change even if the
+consistency check is made. Security analysis is required. See #401.]]
+
+One way a Helper could address this would be to store a digest of the Leader's
+request, indexed by aggregation job ID and round, and refuse to service a
+request for a given aggregation job round unless it matches the previously seen
+request (if any).
+
+As there are several implementation strategies for round skew recovery,
```

Comment:
nit: IMO, this sentence could be dropped -- we don't specify a solution to many implementation problems, and we don't need to call out every instance. (We _could_ also drop the preceding paragraph, as it does specify one possible implementation strategy for the round-skew recovery check.)